ui/page/dexclient/confirm_register_modal.go

```diff
@@ -0,0 +1,136 @@
+	*load.Load
+	modal            *decredmaterial.Modal
+	register         decredmaterial.Button
+	appPassword      decredmaterial.Editor
+	isSending        bool
+	dex              *core.Exchange
+	cert             []byte
+	selectedFeeAsset string
+	confirmed        func()
+}
+
+func newConfirmRegisterModal(l *load.Load, dex *core.Exchange, cert []byte, selectedFeeAsset string) *confirmRegisterModal {
+	md := &confirmRegisterModal{
+		Load:             l,
+		modal:            l.Theme.ModalFloatTitle(),
+		appPassword:      l.Theme.EditorPassword(new(widget.Editor), "App password"),
+		register:         l.Theme.Button("Register"),
+		dex:              dex,
+		cert:             cert,
+		selectedFeeAsset: selectedFeeAsset,
+	}
+
+	md.register.TextSize = values.TextSize12
+	md.register.Background = l.Theme.Color.Primary
+	md.appPassword.Editor.SingleLine = true
+
+	return md
+}
+
+func (md *confirmRegisterModal) ModalID() string {
+	return confirmRegisterModalID
+}
+
+func (md *confirmRegisterModal) Show() {
+	md.ShowModal(md)
+}
+
+func (md *confirmRegisterModal) Dismiss() {
+	md.DismissModal(md)
+}
+
+func (md *confirmRegisterModal) OnDismiss() {
+	md.appPassword.Editor.SetText("")
+}
+
+func (md *confirmRegisterModal) OnResume() {
+	md.appPassword.Editor.Focus()
+}
+
+func (md *confirmRegisterModal) Handle() {
+	if md.register.Button.Clicked() {
+		if md.appPassword.Editor.Text() == "" || md.isSending {
+			return
+		}
+
+		md.isSending = true
+		go func() {
+			_, err := md.Dexc().RegisterWithDEXServer(md.dex.Host,
+				md.cert,
+				int64(md.dex.Fee.Amt),
+				int32(md.dex.Fee.ID),
+				[]byte(md.appPassword.Editor.Text()))
+
+			md.isSending = false
+			if err != nil {
+				md.Toast.NotifyError(err.Error())
+				return
+			}
+
+			md.confirmed()
+			md.Dismiss()
+		}()
+	}
+}
+
+func (md *confirmRegisterModal) Layout(gtx layout.Context) D {
+	w := []layout.Widget{
+		func(gtx C) D {
+			return md.Load.Theme.Label(values.TextSize20, "Confirm Registration").Layout(gtx)
+		},
+		func(gtx C) D {
+			feeAsset := md.dex.RegFees[md.selectedFeeAsset]
+			feeAmt := formatAmount(feeAsset.ID, md.selectedFeeAsset, feeAsset.Amt)
+			txt := fmt.Sprintf("Enter your app password to confirm DEX registration. When you submit this form, %s will be spent from your wallet to pay registration fees.", feeAmt)
+			return md.Load.Theme.Label(values.TextSize14, txt).Layout(gtx)
+		},
+		func(gtx C) D {
+			markets := make([]string, 0, len(md.dex.Markets))
+			for _, mkt := range md.dex.Markets {
+				lotSize := formatAmount(mkt.BaseID, mkt.BaseSymbol, mkt.LotSize)
+				markets = append(markets, fmt.Sprintf("Base: %s\tQuote: %s\tLot Size: %s", strings.ToUpper(mkt.BaseSymbol), strings.ToUpper(mkt.QuoteSymbol), lotSize))
+			}
+			txt := fmt.Sprintf("This DEX supports the following markets. All trades are in multiples of each market's lot size.\n\n%s", strings.Join(markets, "\n"))
+			return md.Load.Theme.Label(values.TextSize14, txt).Layout(gtx)
+		},
```

Comment:
You can use a condition to have this displayed on the `NewPasswordModal()` modal.